catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/AcftbAttribute.java

```diff
@@ -13,123 +13,123 @@
-    IMMEDIATE_SCENE_REQUEST_ID("immediateSceneRequestId",
-            "IMREQID",
-            tre -> TreUtility.getTreValue(tre, "IMREQID")),
-    MISSION_PLAN_MODE("missionPlanMode", "MPLAN", tre -> TreUtility.getTreValue(tre, "MPLAN"));
-
-    private static final String ATTRIBUTE_NAME_PREFIX = "acftb.";
-
-    private String shortName;
-
-    private String longName;
+            tre -> TreUtility.getTreValue(tre, "IMHOSTNO"),
+            BasicTypes.STRING_TYPE,
+            ATTRIBUTE_NAME_PREFIX);
 
-    private Function<Tre, Serializable> accessorFunction;
-
-    private Set<AttributeDescriptor> attributeDescriptors;
-
-    AcftbAttribute(final String longName, final String shortName,
-            final Function<Tre, Serializable> accessorFunction) {
-        this.longName = longName;
-        this.shortName = shortName;
-        this.accessorFunction = accessorFunction;
-        // retrieving metacard attribute descriptors for this attribute to prevent later lookups
-        this.attributeDescriptors = new HashSet<>();
-        this.attributeDescriptors.add(new AttributeDescriptorImpl(
-                ExtNitfUtility.EXT_NITF_PREFIX + ATTRIBUTE_NAME_PREFIX + longName,
-                true, /* indexed */
-                true, /* stored */
-                false, /* tokenized */
-                true, /* multivalued */
-                BasicTypes.STRING_TYPE));
-    }
-
-    AcftbAttribute(String longName, String shortName, Function<Tre, Serializable> accessorFunction,
-            AttributeDescriptor attributeDescriptor, String extNitfName) {
-        this.longName = longName;
-        this.shortName = shortName;
-        this.accessorFunction = accessorFunction;
-        // retrieving metacard attribute descriptors for this attribute to prevent later lookups
-        this.attributeDescriptors = new HashSet<>();
-        this.attributeDescriptors.add(attributeDescriptor);
-        if (StringUtils.isNotEmpty(extNitfName)) {
-            this.attributeDescriptors.add(ExtNitfUtility.createDuplicateDescriptorAndRename(
-                    ATTRIBUTE_NAME_PREFIX + extNitfName,
-                    attributeDescriptor));
-        }
-    }
-
-    @Override
-    public String getLongName() {
-        return this.longName;
-    }
-
-    @Override
-    public String getShortName() {
-        return this.shortName;
+    public static final NitfAttribute<Tre> IMMEDIATE_SCENE_REQUEST_ID = new NitfAttributeImpl<>(
+            "immediateSceneRequestId",
+            "IMREQID",
+            tre -> TreUtility.getTreValue(tre, "IMREQID"),
+            BasicTypes.STRING_TYPE,
+            ATTRIBUTE_NAME_PREFIX);
+
+    public static final NitfAttribute<Tre> MISSION_PLAN_MODE = new NitfAttributeImpl<>(
+            "missionPlanMode",
+            "MPLAN",
+            tre -> TreUtility.getTreValue(tre, "MPLAN"),
+            BasicTypes.STRING_TYPE,
+            ATTRIBUTE_NAME_PREFIX);
+
+    private List<NitfAttribute<Tre>> attributes;
```

Comment:
Could we make this list and `getAttributes()` `static` and initialize the list in a static initializer block? That way the constructor wouldn't have to be `public` and the class wouldn't need to be instantiated just to get the list of attributes.

Another option: instead of the static initializer, could would have this class extend from `NitfAttributeImpl` and have the `AcftbAttribute` constructor add each new constant to the `static` attribute list? That way, if a new constant is added it will automatically be added to the list. No chance of missing one.

Thoughts?

Note: this applies to all the constant classes.